### PR TITLE
Update versions of sonar dependencies

### PR DESCRIPTION
**Pull request by francoismora**
Update sonarqube version to latest and introduce dependency to sonar-plugin-api pointing to latest version.

### CODE REVIEW

**Review by vilchik-elena — APPROVED** [review]

**Review by vilchik-elena — CHANGES_REQUESTED** [review]
Plugin test for SonarLint failed, not sure why. May be there sonarlint API version we use is somehow not compatible

File: its/plugin/tests/pom.xml [hunk 1]
```diff
@@ -30,14 +30,14 @@
...
     </dependency>
-    <dependency>
-      <groupId>org.sonarsource.sonarqube</groupId>
-      <artifactId>sonar-ws</artifactId>
-    </dependency>
     <dependency>
       <groupId>org.sonarsource.sonarlint.core</groupId>
       <artifactId>sonarlint-core</artifactId>
     </dependency>
+    <dependency>
```
Comment (vilchik-elena) [inline its/plugin/tests/pom.xml:37]: @francoismora could you share why this fixes the problem?

Comment (francoismora) [reply its/plugin/tests/pom.xml:37]: A method has been removed in the latest version of the plugin API (`Loggers.setTarget()`) but was still available in the version included in sonarlint. Changing the classpath order fixed the issue, but we decided with @saberduck to upgrade sonarlint instead to have all components aligned to the latest version.

Comment (vilchik-elena) [reply its/plugin/tests/pom.xml:37]: should we revert this change of order then?

**Review by vilchik-elena — APPROVED** [review]
LGTM, just wonder if we can revert the change of order

**Pull request merged**